### PR TITLE
Add DISK Extractor and LightGlue Matcher

### PR DESCRIPTION
**Pull request by travisdriver**

### CODE REVIEW

File: gtsfm/frontend/detector_descriptor/disk.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+from thirdparty.LightGlue.lightglue.disk import DISK
+
+ROOT_PATH = Path(__file__).resolve().parent.parent.parent.parent
+MODEL_WEIGHTS_PATH = (
+    ROOT_PATH / "thirdparty" / "SuperGluePretrainedNetwork" / "models" / "weights" / "superpoint_v1.pth"
+)
+
+
+class DiskDetectorDescriptor(DetectorDescriptorBase):
+    """Superpoint Detector+Descriptor implementation."""
```
Comment (ayushbaid) [inline gtsfm/frontend/detector_descriptor/disk.py:30]: Please correct the docstring

File: gtsfm/frontend/detector_descriptor/disk.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    def __init__(
+        self,
+        max_keypoints: int = 5000,
+        use_cuda: bool = True,
+        weights_path: Union[Path, str] = MODEL_WEIGHTS_PATH,
```
Comment (ayushbaid) [inline gtsfm/frontend/detector_descriptor/disk.py:36]: Where is this argument being used?

File: gtsfm/frontend/matcher/lightglue_matcher.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import torch
+
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.matcher.matcher_base import MatcherBase
+from thirdparty.LightGlue.lightglue.lightglue import LightGlue
+from thirdparty.LightGlue.lightglue.utils import rbd
+
+
+class LightGlueMatcher(MatcherBase):
+    """Implements the SuperGlue matcher -- a pretrained graph neural network using attention."""
```
Comment (ayushbaid) [inline gtsfm/frontend/matcher/lightglue_matcher.py:25]: Please correct the docstring.

File: gtsfm/frontend/matcher/lightglue_matcher.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            "descriptors": torch.from_numpy(descriptors_i2).unsqueeze(0).float().to(device),
+            "image": empty_image_i2.to(device),
+        }
+
+        # Match!
+        with torch.no_grad():
+            matches = self._model({"image0": feats_i1, "image1": feats_i2})
+        feats_i1, feats_i2, matches = [rbd(x) for x in [feats_i1, feats_i2, matches]]  # remove batch dimension
+        matches = matches["matches"].detach().cpu().numpy()  # indices with shape (N, 2)
+        print(matches.shape)
```
Comment (ayushbaid) [inline gtsfm/frontend/matcher/lightglue_matcher.py:92]: Remove print

**Review by ayushbaid — APPROVED** [review]
Please fix the minor comments before merging.

File: gtsfm/frontend/matcher/lightglue_matcher.py (hunk 4 above)
Comment (travisdriver) [reply gtsfm/frontend/matcher/lightglue_matcher.py]: done

File: gtsfm/frontend/matcher/lightglue_matcher.py (hunk 3 above)
Comment (travisdriver) [reply gtsfm/frontend/matcher/lightglue_matcher.py]: done

File: gtsfm/frontend/detector_descriptor/disk.py (hunk 2 above)
Comment (travisdriver) [reply gtsfm/frontend/detector_descriptor/disk.py]: done

File: gtsfm/frontend/detector_descriptor/disk.py (hunk 1 above)
Comment (travisdriver) [reply gtsfm/frontend/detector_descriptor/disk.py]: done

**Comment by travisdriver** [comment]
# Skydio-32 reconstruction results

### DISK + LightGlue

Screen Shot 2023-07-25 at 10 20 56 AM

### SuperPoint + SuperGlue

Screen Shot 2023-07-25 at 10 21 35 AM

File: gtsfm/frontend/detector_descriptor/disk.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        """Configures the object.
+
+        Args:
+            max_keypoints: max keypoints to detect in an image.
+            use_cuda (optional): flag controlling the use of GPUs via CUDA. Defaults to True.
+            weights_path (optional): Path to the model weights. Defaults to MODEL_WEIGHT_PATH.
+        """
+        super().__init__(max_keypoints=max_keypoints)
+        self._use_cuda = use_cuda
+        # self._config = {"weights_path": weights_path}
```
Comment (johnwlambert) [inline gtsfm/frontend/detector_descriptor/disk.py:40]: can we remove this line?

File: .github/workflows/benchmark-self-hosted.yml [hunk 6]
```diff
@@ -19,7 +19,16 @@ jobs:
       matrix:
         config_dataset_info: [
             # config             dataset       lookahead   img-extension  source  loader         max-res  share-intrinsics
-            [skydio_front_end,   skydio-501,   15,         jpg,           wget,   colmap-loader, 600,     true,],
+            [
+              skydio_front_end,
```
Comment (johnwlambert) [inline .github/workflows/benchmark-self-hosted.yml:23]: nit: i think was a bit more readable before, but up to you

Comment (travisdriver) [reply .github/workflows/benchmark-self-hosted.yml:23]: it autoformatted, will change back, thanks for catching

Comment (travisdriver) [reply .github/workflows/benchmark-self-hosted.yml]: done

**Comment by travisdriver** [comment]
Slight improvement of SuperPoint + LightGlue over SuperPoint + SuperGlue. Think the results are convincing enough to make it the new default.

newplot (10)
[visual_comparison_dashboard.html.zip](https://github.com/borglab/gtsfm/files/12175789/visual_comparison_dashboard.html.zip)

**Pull request merged**